universal-application-tool-0.0.1/test/app/BaseBrowserTest.java

```diff
@@ -165,4 +167,30 @@ protected void removeQuestionsToProgram(String programName, String... questions)
       assertThat(browser.$("#questionBankQuestions").$("li").textContents()).contains(question);
     }
   }
+
+  protected void fillInput(String name, String text) {
+    browser.$("input", withName(name)).fill().with(text);
+  }
+
+  protected String getInputValue(String name) {
+    return browser.$("input", withName(name)).first().value();
+  }
+
+  protected void fillTextArea(String name, String text) {
+    WebElement element =
+        browser.getDriver().findElement(By.cssSelector("textarea[name=" + name + "]"));
+    element.clear();
```

Comment:
question (non-blocking): Why is `fillTextArea` implemented differently from `fillInput`? Given their names I'd expect them to behave the same: either both clear before filling in, or both not.

Reply:
I fought this a bunch yesterday. Basically `textarea` gets and sets the value differently from `input`. For input it's an `attribute` but for textarea it's a `property` so it does not play nicely with Selenium. I didn't feel like I was encountering issues with the fill portion of textarea, but was running into a ton of issues trying to pull the value back out. A bunch of searching suggested that this was the proper way to fill the textarea. It may not be necessary, and it didn't solve the problem of not being able to get the value, but I kept it this way since it does seem cleaner.

**TLDR:** I wrote it this way because I was trying to solve a related issue. It didn't solve the related issue but I kept it this way anyway.